dev/com.ibm.ws.jbatch.security/src/com/ibm/ws/jbatch/security/impl/WSBatchAuthServiceImpl.java

```diff
@@ -265,8 +367,37 @@ private void authorizedJobStopRestart(String submitter, long instanceId)
     																new Object[] {getRunAsUser(), instanceId},
     																"CWWKY0302W: User {0} is not authorized to perform batch operations associated with job instance {1}.") );
     		}
-    	}else if (this.isMonitor(runAsSubject())) {
+    	} else if (this.isGroupAdmin(runAsSubject())) {
+			try {	
+				//listOfGroupsForJobID = persistenceManagerService.getGroupNamesForJobID(instanceId);
+				listOfGroupsForJobID = persistenceManagerService.getJobInstance(instanceId).getGroupName();
+				listOfGroupsForSubject = getSubjectGroups(runAsSubject());
+				// if (!listOfGroupsForJobID.isEmpty()) { //there are groups
+				// associated with this jobID
+				if (subjectInGroups(listOfGroupsForSubject, listOfGroupsForJobID)) {
+					// allow access
+					logger.finer("group security: access would be allowed");
+					
+				} else {
+					// user not the submitter and is not in any groups listed - disallow access
+					logger.fine(getFormattedMessage("USER_GROUP_UNAUTHORIZED_JOB_INSTANCE",
+							new Object[] {instanceId, getRunAsUser(), constructGroupListForAuthFailString(listOfGroupsForJobID)},
+							"CWWKY0305W: Access to job instance {1} denied.  The job has an operation group name defined and the user {2} has batchGroupMonitor or batchGroupAdmin authority but is not a member of the any appropriate group {3}."));
+					throw new JobSecurityException( getFormattedMessage( "USER_UNAUTHORIZED_NO_BATCH_ROLES",
+                            new Object[] { getRunAsUser() },
+                            "CWWKY0303W: User {0} is not authorized to perform any batch operations." ) );
+				}
+        	} catch (NoSuchJobInstanceException nsjex) {
+        		//no groups associated with jobid - disallow access
+        		logger.finer("group security: no jobIDGroup entries found in table - access to this job disallowed");
+                throw new JobSecurityException( getFormattedMessage( "USER_UNAUTHORIZED_NO_BATCH_ROLES",
+                        new Object[] { getRunAsUser() },
+                        "CWWKY0303W: User {0} is not authorized to perform any batch operations." ) );
+        	}
+    	} else if (this.isMonitor(runAsSubject())) {
     		throw new JobSecurityException("Current user " + this.getRunAsUser()+ " with role batch_monitor is not authorized to stop or restart jobs.");
+    	}else if (this.isGroupMonitor(runAsSubject())) {
```

Comment:
This could be collapsed into the other one:   `if (isMonitor() || isGroupMonitor() ) {   throw new ...."Current user ... with role monitor or group monitor is not authorized...`    Maybe better yet, just switch them both over to:  USER_UNAUTHORIZED_JOB_INSTANCE .  Then we're still differentiating btw USER_UNAUTHORIZED_NO_BATCH_ROLES and this case.

Reply:
done